solutions/CF Save the Nature.mdx

```diff
@@ -0,0 +1,89 @@
+---
+id: cf-1223C
+source: CF
+title: Save the Nature
+author: Jesse Choe, Kevin Sheng
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/66993)
+
+Binary search on the answer for the minimum number of tickets, which we call $t$, to sell to make the total ecological contribution to at least $k$. 
+
+To check if a certain number of tickets reaches a minimal contribution of $k$, we can greedily observe that in order to minimize the number of tickets sold, we should maximize the total revenue from selling these tickets. If, however, it is impossible to sell at least $k$ tickets after selling all tickets, then output $-1$.
+
+Additionally, the problem indicates that only $x\%$ of the $a$-th ticket and $y\%$ of the $b$-th ticket could contribute to the total ecological contribution. Thus, we can iterate over every $a$-th ticket and $b$-th ticket at most $t$ and add $x\%$ to the total percentages contributing to an ecological program for each $a$-th ticket and $y\%$ for each $b$-th ticket, which we can store in $pct$.
```

Comment:
what

Reply:
> Thus, we can iterate over every $a$-th ticket and $b$-th ticket at most $t$

what does this mean
and how are we going to order the tickets even
not to mention handling the combo tickets

Reply:
> and how are we going to order the tickets even

sorting in descending order. Oh shit maybe i shld put that :(.

Reply:
no you already said that
the thing is we already know how many x% tickets and y% tickets we have
but 1. how we're going to distribute them isn't really clear and 2. how the combo tickets are handled isn't mentioned at all

Reply:
probably phrase it more like: we can precompute the multipliers over all $t$ tickets. then greedily match the largest multiplier w/ highest value ticket.

Reply:
but what do we mean by "multipliers"? maybe we should mention explicitly that there's 3 types of multiples- x, y, and x+y